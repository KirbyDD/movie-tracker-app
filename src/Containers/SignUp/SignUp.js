import React, { Component } from 'react'
import { connect } from 'react-redux' 

class SignUp extends Component {
  constructor() {
    super();
    this.state = {
      
    }
  }
  render() {
    return (
      <div>
        
      </div>
    )
  }
}

export default SignUp
