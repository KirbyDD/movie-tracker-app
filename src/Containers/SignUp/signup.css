.signup-btn {
  height: 40px;
  width: 80%;
  border-radius: 5px;
  background: rgba(1, 210, 120, 0.753);
  cursor: pointer;
  font-size: 16px;
  outline:none;
  margin-top: 20px;
}

.signup-btn:hover {
  background: #01D277;
}

.signup-form {
  height: 660px;
  width: 450px;
  border-radius: 7px;
  margin-left: auto;
  margin-right: auto;
  background-image: linear-gradient(rgba(0, 0, 0, 0.822), rgba(0, 0, 0, 0.733));
  box-shadow: 0 0 10px 5px rgba(255, 255, 255, 0.027);
  margin-top: 120px;
  display: flex;
  flex-direction: column;
}

.signup-input-btn-container {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  height: 500px;
  margin-top: 40px;
}