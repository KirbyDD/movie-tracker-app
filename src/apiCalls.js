export const fetchFilms = () => {
  let url = 'https://api.themoviedb.org/3/movie/now_playing?api_key=faeff26a101acfb04b4ebb285bac57bf&language=en-US&page=1'
   return fetch(url)
     .then(response => response.json())
     .then(data => data)
};

export const fetchGenre = () => {
  let url = 'https://api.themoviedb.org/3/genre/movie/list?api_key=faeff26a101acfb04b4ebb285bac57bf&language=en-US'
   return fetch(url)
     .then(response => response.json())
     .then(data => data)
};



