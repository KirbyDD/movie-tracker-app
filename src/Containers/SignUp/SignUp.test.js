import React from 'react'
import { shallow }  from 'enzyme'
import SignUp from '../SignUp/SignUp';
import { postNewUser } from '../../apiCalls';

describe('SignUp', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<SignUp />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should have a default state', () => {
    const expectedState = {
      name: '',
      email: '',
      password: '',
      password2: '',
      currentUser: '',
      error: ''
    }
    expect(wrapper.state()).toEqual(expectedState)
  })

  it('should be able to handle input change based on events', () => {
    const event = {target: {name: 'email', value: 'ryan@flachman.net'}}
    wrapper.instance().handleInput(event)
    const expectedEmail = 'ryan@flachman.net'
    expect(wrapper.state().email).toEqual(expectedEmail)
  })

  it('Should have a method to compare two passwords and return true/false', () => {
    const expected = true;
    const result = wrapper.instance().checkPassword('password', 'password')
    expect(result).toEqual(expected)
  })

  it('should have a handleSignUp method that calls of postNewUser api call', () => {
    wrapper.instance().handleSignUp = jest.fn();
    wrapper.find('.signup-btn').simulate('click', {
      preventDefault: () => {}
    })
    expect(wrapper.instance().handleSignUp).toHaveBeenCalled();
  })

  it('should call handleInput when change happen', () => {
    wrapper.instance().handleInput = jest.fn();
    wrapper.find('#signup-name').simulate('change', {
      target: {name: 'name', value: 'a'}
    })
    expect(wrapper.instance().handleInput).toHaveBeenCalled();
    wrapper.find('#signup-email').simulate('change', {
      target: {name: 'email', value: 'a'}
    })
    expect(wrapper.instance().handleInput).toHaveBeenCalled();
    wrapper.find('#signup-password-1').simulate('change', {
      target: {name: 'password', value: 'a'}
    })
    expect(wrapper.instance().handleInput).toHaveBeenCalled();
    wrapper.find('#signup-password-2').simulate('change', {
      target: {name: 'password2', value: 'a'}
    })
    expect(wrapper.instance().handleInput).toHaveBeenCalled();
  })
})