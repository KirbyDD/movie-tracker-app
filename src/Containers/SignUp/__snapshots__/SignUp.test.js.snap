// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should match the snapshot 1`] = `
<div
  className="signin-container"
>
  <form
    className="signup-form"
  >
    <h2
      className="sigin-header"
    >
      Sign Up
    </h2>
    <div
      className="signup-input-btn-container"
    >
      <div
        className="input-container"
      >
        <label
          htmlFor="signup-name"
        >
          Name
        </label>
        <input
          id="signup-name"
          name="name"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="input-container"
      >
        <label
          htmlFor="signup-email"
        >
          Email
        </label>
        <input
          id="signup-email"
          name="email"
          onChange={[Function]}
          type="email"
          value=""
        />
      </div>
      <div
        className="input-container"
      >
        <label
          htmlFor="signup-password-1"
        >
          Password
        </label>
        <input
          id="signup-password-1"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
      <div
        className="input-container"
      >
        <label
          htmlFor="signup-password-2"
        >
          Re-Enter Password
        </label>
        <input
          id="signup-password-2"
          name="password2"
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
      <button
        className="signup-btn"
        onClick={[Function]}
      >
        Sign Up
      </button>
    </div>
  </form>
</div>
`;
